notification/events.go

```diff
@@ -28,6 +41,38 @@ func RegisterEvents(ctx context.Context) {
 	events.RegisterAsyncHandler(sendNotifications, 1, 5, api.EventNotificationSend)
 }
 
+func getOrCreateRateLimiter(ctx context.Context, notificationID string) (*sw.Limiter, error) {
+	rateLimitersLock.Lock()
```

Comment:
Can we use a sync.Map instead - we don't want to  keep a lock while perform db operations.

Reply:
@moshloop we want to lock the db operation in here though. For a sudden burst of event, we only want to make one db query and create one rate limiter for a notification.

This only happens once throughout the lifetime of the process as we'll then hit the cache.